cf-agent/verify_files.c

```diff
@@ -755,19 +795,36 @@ PromiseResult ScheduleEditOperation(EvalContext *ctx, char *filename, Attributes
         }
     }
 
-
-    if (a.edit_template)
+    if (strcmp("cfengine", a.template_method) == 0)
     {
-        Log(LOG_LEVEL_VERBOSE, "Rendering '%s' using template '%s' with method '%s'",
-            filename, a.edit_template, a.template_method ? a.template_method : "cfengine");
-        if (!a.template_method || strcmp("cfengine", a.template_method) == 0)
+        if (a.edit_template)
         {
+            Log(LOG_LEVEL_VERBOSE, "Rendering '%s' using template '%s' with method '%s'",
+                filename, a.edit_template, a.template_method);
+
             PromiseResult render_result = RenderTemplateCFEngine(ctx, pp, args, a, edcontext);
             result = PromiseResultUpdate(result, render_result);
         }
-        else if (strcmp("mustache", a.template_method) == 0)
+    }
+    else if (strcmp("mustache", a.template_method) == 0)
+    {
+        if (a.edit_template)
         {
-            PromiseResult render_result = RenderTemplateMustache(ctx, pp, a, edcontext);
+            Log(LOG_LEVEL_VERBOSE, "Rendering '%s' using template '%s' with method '%s'",
+                filename, a.edit_template, a.template_method);
+
+            PromiseResult render_result = RenderTemplateMustacheFromFIle(ctx, pp, a, edcontext);
+            result = PromiseResultUpdate(result, render_result);
+        }
+    }
+    else if (strcmp("inline_mustache", a.template_method) == 0)
+    {
+        if (a.edit_template_string)
+        {
+            Log(LOG_LEVEL_VERBOSE, "Rendering '%s' with method '%s'",
+                filename, a.template_method);
```

Comment:
what is the filename here?

Reply:
That is the generated filename. The promiser ;-)